src/workspace.ts

```diff
@@ -95,18 +95,32 @@ export async function patchSettings(patch: any, askText: string): Promise<boolea
 
     // Patch has been applied in-memory, and now we have to update the file (settings.json).
     // Ask for permission.
-    let allow = await presenter.askYesNo(askText);
+    let allow = await askForPermission();
     if (!allow) {
         return false;
     }
 
     let content = JSON.stringify(settings, null, 4);
     fs.writeFileSync(filePath, content);
-    Feedback.info("Updated settings.json.");
+    Feedback.info(`Updated ${filePath}.`);
 
     return true;
 }
 
+export async function askPatchSettings(patch: any, filePath: string, askText: string): Promise<boolean> {
```

Comment:
Here, the name of the function hides the fact that an important operation will be executed after asking. Perhaps `patchSettingsAfterAsking` / `promptThenPatchSettings` / `patchSettingsWithPrompt`?

Reply:
Done.